cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryRedirect.ts

```diff
@@ -1,20 +1,58 @@
-import { CloudFrontRequestHandler } from 'aws-lambda';
+import {CloudFrontRequestEvent, CloudFrontRequestHandler, CloudFrontResultResponse, Context} from 'aws-lambda';
 import { CloudFrontRequest } from 'aws-lambda/common/cloudfront';
+import axios from 'axios';
+import assert from "assert";
+import { CatalogueResultsList } from './catalogue'
 
-export const request: CloudFrontRequestHandler = (event, context, callback) => {
+async function apiFetch(path: string) {
+  return axios.get(
+    'https://api.wellcomecollection.org/catalogue/v2' + path
+  )
+}
+
+// TODO: Account for multiple results by looking for matching sierra entry
+// TODO: Account for situations where results overflow page
+async function getWork(bNumber: string) {
+  const apiPath = `/works?identifiers=${bNumber}`
+  const results = await apiFetch(apiPath)
+
+  const resultList = results.data as CatalogueResultsList
+  assert(resultList.results.length >= 1, "No matching Catalogue API results found")
+
+  return resultList.results[0]
+}
+
+// TODO: Properly test for all these cases
+function getBnumberFromPath(path: string) {
+  const splitPath = path.split("/")
+  const bNumberRegexp = /^b[0-9]{8}/
```

Comment:
This is wrong – b numbers can end with an `x` as the check digit.

Reply:
Also, it would be handy to know what sort of path you expect this to match.

Possibly move the extraction logic into a separate file? It's just string parsing, could be managed/tested on its own.

Reply:
[Assuming that this is where we'll do it though](https://github.com/wellcomecollection/platform-infrastructure/pull/116/files#diff-cad288977e3edeafbbc2ad132d4fd147a6762ba2623c35f9fc6379e3d638d2d3R38)

Reply:
>This is wrong – b numbers can end with an x as the check digit.

I'm not sure if they can in this context, this is redirecting a specific pattern from the wellcomelibrary site. I'm not sure that there would be any links of this type pointing at un-check-digited bnumbers (I couldn't find any when I went a-googling).

Reply:
> Possibly move the extraction logic into a separate file? It's just string parsing, could be managed/tested on its own. 

👍

Reply:
> I'm not sure if they can in this context, this is redirecting a specific pattern from the wellcomelibrary site.

If so, maybe better to match on 7 digits? 8 digits implies the presence of a check digit.

Reply:
I would also say it's odd practise in terms of it not really being errors, but rather known states of paths we would redirect.

The more common approach would be to return either `undefined` or a reason if that's needed later for logging, but you could just do the logging here.

Reply:
I couldn't find any links of that form (7 digits), but I don't think there's any harm in being flexible (and possibly some in not).